src/test/tpc/tpcds_db_generator_test.cpp

```diff
@@ -0,0 +1,157 @@
+        table_generator.generate_catalog_sales_and_returns(rows_to_check);
+    EXPECT_TABLE_EQ_ORDERED(catalog_sales_table, load_csv("catalog_sales.csv"));
+    EXPECT_TABLE_EQ_ORDERED(catalog_returns_table, load_csv("catalog_returns.csv"));
+    EXPECT_TABLE_EQ_ORDERED(table_generator.generate_customer_address(rows_to_check), load_csv("customer_address.csv"));
+    EXPECT_TABLE_EQ_ORDERED(table_generator.generate_customer(rows_to_check), load_csv("customer.csv"));
+    EXPECT_TABLE_EQ_ORDERED(table_generator.generate_customer_demographics(rows_to_check),
+                            load_csv("customer_demographics.csv"));
+    EXPECT_TABLE_EQ_ORDERED(table_generator.generate_date(rows_to_check), load_csv("date_dim.csv"));
+    EXPECT_TABLE_EQ_ORDERED(table_generator.generate_household_demographics(rows_to_check),
+                            load_csv("household_demographics.csv"));
+    EXPECT_TABLE_EQ_ORDERED(table_generator.generate_income_band(rows_to_check), load_csv("income_band.csv"));
+    EXPECT_TABLE_EQ_ORDERED(table_generator.generate_inventory(rows_to_check), load_csv("inventory.csv"));
+    EXPECT_TABLE_EQ_ORDERED(table_generator.generate_item(rows_to_check), load_csv("item.csv"));
+    EXPECT_TABLE_EQ_ORDERED(table_generator.generate_promotion(rows_to_check), load_csv("promotion.csv"));
+    EXPECT_TABLE_EQ_ORDERED(table_generator.generate_reason(rows_to_check), load_csv("reason.csv"));
+    EXPECT_TABLE_EQ_ORDERED(table_generator.generate_ship_mode(rows_to_check), load_csv("ship_mode.csv"));
+    EXPECT_TABLE_EQ_ORDERED(table_generator.generate_store(rows_to_check), load_csv("store.csv"));
+    const auto [store_sales_table, store_returns_table] =
+        table_generator.generate_store_sales_and_returns(rows_to_check);
+    EXPECT_TABLE_EQ_ORDERED(store_sales_table, load_csv("store_sales.csv"));
+    EXPECT_TABLE_EQ_ORDERED(store_returns_table, load_csv("store_returns.csv"));
+    EXPECT_TABLE_EQ_ORDERED(table_generator.generate_time(rows_to_check), load_csv("time_dim.csv"));
+    EXPECT_TABLE_EQ_ORDERED(table_generator.generate_warehouse(rows_to_check), load_csv("warehouse.csv"));
+    EXPECT_TABLE_EQ_ORDERED(table_generator.generate_web_page(rows_to_check), load_csv("web_page.csv"));
+    const auto [web_sales_table, web_returns_table] = table_generator.generate_web_sales_and_returns(rows_to_check);
+    EXPECT_TABLE_EQ_ORDERED(web_sales_table, load_csv("web_sales.csv"));
+    EXPECT_TABLE_EQ_ORDERED(web_returns_table, load_csv("web_returns.csv"));
+    EXPECT_TABLE_EQ_ORDERED(table_generator.generate_web_site(rows_to_check), load_csv("web_site.csv"));
+  }
+}
+
+TEST(TpcdsTableGeneratorTest, GenerateAndStoreRowCounts) {
+  /**
+ * Check whether all TPC-DS tables are created by the TpcdsTableGenerator and added to the StorageManager.
+ * Then check whether the row count is correct for all tables.
+ */
+
+  EXPECT_FALSE(StorageManager::get().has_table("call_center"));
```

Comment:
There's nothing wrong with it. But you could also use `StorageManager::get().tables().size()` check for `0` and a comment that you are checking if the StorageManager is empty.

Reply:
I remember Moritz told me not to use loops in tests, because they lead to unclear error messages. You only get to know the line, but not in which iteration of the loop the error happened.

Reply:
You can simply do `EXPECT_EQ(...) << name` so that you can see the table_name in question. I wouldn't say that expects in loops are generally bad, but it makes sense to make identifying the issue easy.

Reply:
I added a SCOPED_TRACE here too, so that should be fine.